Validation/HGCalTriggerPrimitives/plugins/HGCalTrigValidator.cc

```diff
@@ -0,0 +1,309 @@
+
+  protected:
+    void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
+
+  private:
+// ----------member data ---------------------------
+    //histogram tc related
+    MonitorElement *h_tc_n, *h_tc_id, *h_tc_subdet, *h_tc_zside, *h_tc_layer;
+    MonitorElement *h_tc_wafer, *h_tc_wafertype, *h_tc_cell, *h_tc_data;
+    MonitorElement *h_tc_mipPt, *h_tc_pt, *h_tc_simenergy, *h_tc_x, *h_tc_y; 
+    MonitorElement *h_tc_z, *h_tc_energy, *h_tc_eta, *h_tc_phi;
+
+    //histogram cl related
+    MonitorElement *h_cl_n, *h_cl_mipPt, *h_cl_pt; 
+    MonitorElement *h_cl_energy, *h_cl_eta, *h_cl_phi;
+    MonitorElement *h_cl_layer, *h_cl_subdet, *h_cl_cells_n;
+
+    //histogram multicl related
+    MonitorElement *h_cl3d_n, *h_cl3d_pt, *h_cl3d_energy;
+    MonitorElement *h_cl3d_eta, *h_cl3d_phi, *h_cl3d_clusters_n;
+    // cluster shower shapes
+    MonitorElement *h_cl3d_showerlength, *h_cl3d_coreshowerlength, *h_cl3d_firstlayer;
+    MonitorElement *h_cl3d_maxlayer, *h_cl3d_seetot, *h_cl3d_seemax;
+    MonitorElement *h_cl3d_spptot, *h_cl3d_sppmax, *h_cl3d_szz;
+    MonitorElement *h_cl3d_srrtot, *h_cl3d_srrmax, *h_cl3d_srrmean;
+    MonitorElement *h_cl3d_emaxe, *h_cl3d_bdteg, *h_cl3d_quality;
+
+    //histogram tower related
+    MonitorElement *h_tower_n, *h_tower_pt, *h_tower_energy;
+    MonitorElement *h_tower_eta, *h_tower_phi, *h_tower_etEm;
+    MonitorElement *h_tower_etHad, *h_tower_iEta, *h_tower_iPhi;
+
+
+    edm::EDGetToken trigger_cells_token_;
+    edm::EDGetToken clusters_token_;
+    edm::EDGetToken multiclusters_token_; 
+    edm::EDGetToken towers_token_;
+   
+   std::unique_ptr<HGCalTriggerClusterIdentificationBase> id_;
+   
+private:
+    int tc_n;
+    int cl_n;
+    int cl3d_n;
+    int tower_n;
+    
+    HGCalTriggerTools triggerTools_;
+};
+
+HGCalTrigValidator::HGCalTrigValidator(const edm::ParameterSet& iConfig){
+
+  trigger_cells_token_ = consumes<l1t::HGCalTriggerCellBxCollection>(iConfig.getParameter<edm::InputTag>("TriggerCells"));
+  clusters_token_ = consumes<l1t::HGCalClusterBxCollection>(iConfig.getParameter<edm::InputTag>("Clusters"));
+  multiclusters_token_ = consumes<l1t::HGCalMulticlusterBxCollection>(iConfig.getParameter<edm::InputTag>("Multiclusters"));
+  towers_token_ = consumes<l1t::HGCalTowerBxCollection>(iConfig.getParameter<edm::InputTag>("Towers"));
+  
+  id_.reset( HGCalTriggerClusterIdentificationFactory::get()->create("HGCalTriggerClusterIdentificationBDT") );
+  id_->initialize(iConfig.getParameter<edm::ParameterSet>("EGIdentification"));
+
+}
+
+HGCalTrigValidator::~HGCalTrigValidator(){
+}
+
+void HGCalTrigValidator::bookHistograms(DQMStore::IBooker &iBooker, edm::Run const &, edm::EventSetup const &) {
+
+  //initiating histograms
+  // trigger cells 
+  h_tc_energy = iBooker.book1D("tc_energy","trigCell energy; energy", 70, 0, 70);
+  h_tc_eta = iBooker.book1D("tc_eta","trigCell eta; eta", 8, -4, 4);
```

Comment:
The HGCAL goes up to eta=3, not 4. Also 8 bins is really too small; should have more.